src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt

```diff
@@ -390,6 +390,58 @@ namespace <#= @namespace #>
             return new EdmCollectionExpression(records);
 <#+
         }
+        else if (term.Type.Definition.TypeKind == EdmTypeKind.TypeDefinition || term.Type.Definition.TypeKind == EdmTypeKind.Primitive)
+        {
+            var fullTypeName = actualType.AsActualType().FullTypeName();
+            var fullName = term.FullName();
+            var name = fullName.Substring(fullName.LastIndexOf('.') + 1);
+            var camelName = $"{CamelCase(name)}";
+            if (fullTypeName == "Edm.String")
+            {
+#>
+            return new EdmStringConstant(_<#= camelName #>);
+<#+
+            }
+            else if (fullTypeName == "Edm.PrimitiveType")
+            {
+#>
+            if (_<#= camelName #> != null)
```

Comment:
issue: Testing for null implies a primitive type can be null, if so, this method will not compile.
suggestion: Add an `else` for when `_<#= camelName #> == null`

Reply:
#Resolved

Reply:
#resolved